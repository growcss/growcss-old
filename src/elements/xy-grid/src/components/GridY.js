//@flow
import React, { Component } from "react";
import { XYGridElement } from '../styled/XYGridElement';
import type { GuttersType } from '../types';
import { Gutters as DefaultGutters } from './Gutters';

type GridYType = {
  children?: any,
  gutterType: string,
  gutterSizes: string | number | GuttersType,
  alignX: string,
  alignY: string,
};

export default class GridY extends Component<GridYType>
{
  static defaultProps = {
    gutters: DefaultGutters,
    alignX: 'left',
  };

  render() {
    const {
      children,
      gutterSizes,
      gutterType,
      alignX,
      alignY,
    } = this.props;

    const cells = React.Children.map(children, (thisArg) => {
      return React.cloneElement(
        thisArg,
        {
          vertical: false,
          gutterType,
          gutterSizes
        }
      );
    });

    return (
      <XYGridElement direction='vertical' gutterSizes={gutterSizes} alignX={alignX} alignY={alignY} wrap>
        {cells}
      </XYGridElement>
    );
  }
}
