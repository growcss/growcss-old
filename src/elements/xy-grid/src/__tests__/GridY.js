//@flow
import React from 'react';
import GridY from '../components/GridY';
import 'jest-styled-components'

test('the style output of GridY', () => {
  const wrapper = shallow(<GridY>test</GridY>);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('display', 'flex');
  expect(wrapper).toHaveStyleRule('flex-flow', 'column wrap');
  expect(wrapper).toHaveStyleRule('-webkit-box-orient', 'vertical');
  expect(wrapper).toHaveStyleRule('-webkit-box-direction', 'normal');
});
