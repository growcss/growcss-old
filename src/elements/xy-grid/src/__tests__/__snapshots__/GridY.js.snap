// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the style output of GridY 1`] = `
<XYGridElement
  direction="vertical"
  wrap={true}
/>
`;
